index.js

```diff
@@ -17,29 +17,30 @@ module.exports = function kindOf(val) {
   if (val === true || val === false || val instanceof Boolean) {
     return 'boolean';
   }
+
   if (typeof val !== 'object') {
+    // here: function, symbol, string (literal), number (literal)
     return typeof val;
   }
   if (Array.isArray(val)) {
     return 'array';
   }
-
-  var type = toString.call(val);
-
-  if (val instanceof RegExp || type === '[object RegExp]') {
+  if (val instanceof RegExp) {
     return 'regexp';
   }
-  if (val instanceof Date || type === '[object Date]') {
+  if (val instanceof Date) {
```

Comment:
Very thanks, looks great. But.
I dont see problem with instanceof for Date. Some cases from somewhere?